components/director/internal/domain/apptemplate/resolver.go

```diff
@@ -763,50 +773,153 @@ func validateAppTemplateNameBasedOnProvider(name string, appInput *graphql.Appli
+			return errors.Errorf("App Template with %q label has a missing %q label in the applicationInput", regionLabelKey, regionLabelKey)
+		}
 
-	for _, key := range distinguishLabelKeys {
-		if value, exists := labels[key]; exists {
-			appTemplateDistinguishLabels[key] = value
+		if _, err := extractRegionPlaceholder(appTemplateInput.Placeholders); err != nil {
+			return err
 		}
 	}
 
-	for labelKey, labelValue := range appTemplateDistinguishLabels {
-		msg := fmt.Sprintf("%q: %q", labelKey, labelValue)
+	productLabelArr, ok := productLabel.([]interface{})
+	if !ok {
+		return errors.Errorf("could not parse %q label for application template", r.appTemplateProductLabel)
+	}
 
-		filters := []*labelfilter.LabelFilter{
-			labelfilter.NewForKeyWithQuery(labelKey, fmt.Sprintf("\"%s\"", labelValue)),
+	log.C(ctx).Infof("Getting application template for labels %q: %q", r.appTemplateProductLabel, productLabel)
+	appTemplates, err := r.appTemplateSvc.ListByFilters(ctx, r.buildProductLabelFilter(productLabelArr))
+	if err != nil {
+		if apperrors.IsNotFoundError(err) {
+			log.C(ctx).Infof("There are no application templates for the filter. Proceeding.")
+			return nil
 		}
+		return errors.Wrapf(err, "while getting Application Template for labels %q: %q", r.appTemplateProductLabel, productLabel)
+	}
 
-		if regionExists {
-			if _, ok := region.(string); !ok {
-				return errors.Errorf("%s label should be string", regionLabelKey)
+	existingRegions := make([]string, 0)
+	for _, appTemplate := range appTemplates {
+		regionLabel, err := r.labelSvc.GetByKey(ctx, "", model.AppTemplateLabelableObject, appTemplate.ID, regionLabelKey)
+		if err != nil {
+			if apperrors.IsNotFoundError(err) {
+				continue
 			}
-			filters = append(filters, labelfilter.NewForKeyWithQuery(regionLabelKey, fmt.Sprintf("\"%s\"", region)))
-			msg += fmt.Sprintf(" and %q: %q", regionLabelKey, region)
+			return errors.Wrapf(err, "while getting %q label for Application Template", regionLabelKey)
 		}
 
-		log.C(ctx).Infof("Getting application template for labels %s", msg)
-		appTemplate, err := r.appTemplateSvc.GetByFilters(ctx, filters)
-		if err != nil && !apperrors.IsNotFoundError(err) {
-			return errors.Wrap(err, fmt.Sprintf("Failed to get application template for labels %s", msg))
+		existingRegions = append(existingRegions, regionLabel.Value.(string))
+	}
+
+	if len(appTemplates) > 0 {
+		if len(existingRegions) == 0 {
+			return errors.Errorf("Application Template with %q label is global and already exists", r.appTemplateProductLabel)
 		}
 
-		if appTemplate != nil {
-			errMsg := fmt.Sprintf("Cannot have more than one application template with labels %s", msg)
-			log.C(ctx).Error(errMsg)
-			return errors.New(errMsg)
+		if len(existingRegions) > 0 && !isRegionalAppTemplate {
+			return errors.Errorf("Application Template with %q label is regional. The Application Template should contain a %q label", r.appTemplateProductLabel, regionLabelKey)
```

Comment:
Maybe the error message is meant to be about the input application template, but the meaning of the text is a bit confusing. Some variant of the message can be "Existing application template with %q  label is regional. The input application template should contain a %q label".
Tests should be adjusted also.